app.rb

```diff
@@ -0,0 +1,112 @@
+require_relative 'person'
+require_relative 'classroom'
+require_relative 'student'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @books = []
+    @persons = []
+    @rentals = []
+  end
+
+  def list_all_books
+    if @books.empty?
+      puts "\nNo books yet"
+    else
+      puts "\nBooks:"
+      @books.each_with_index { |book, index| puts "#{index}: #{book.title} by #{book.author}" }
+    end
+  end
+
+  def list_all_persons
+    if @persons.empty?
+      puts "\nNo people yet"
+    else
+      @persons.each_with_index do |person, index|
+        puts "#{index}: [#{person.class}] name: #{person.name} ID: #{person.id} Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_a_person
+    loop do
+      print 'Would you like to create a student (1) or a teacher (2)? [Please enter the corresponding number]:'
+      option = gets.chomp.strip.to_i
+
+      case option
+      when 1
+        create_a_student
+        break
+      when 2
+        create_a_teacher
+        break
+      else
+        puts 'Option unknown'
+      end
+    end
+  end
+
+  def create_a_teacher
+    print 'Name: '
+    age = gets.chomp.to_i
+    print 'Age: '
+    name = gets.chomp
+    print 'Specialisation: '
+    specialization = gets.chomp
+    puts 'Teacher created successfully '
+    @persons.push(Teacher.new(name, age, specialization, parent_permission: true))
+  end
+
+  def create_a_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Classroom: '
+    classroom = gets.chomp
+    Classroom.new(classroom)
+    print 'Has parents permission ? [y/n]: '
+    parent_permission = gets.chomp.downcase == 'y'
+    puts 'Student created successfully '
+    @persons.push(Student.new(age, classroom, name, parent_permission: parent_permission))
+  end
```

Comment:
According to the [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) The parameters to create a student are: `Age`, `Name`, `Has parent permission?`

![image](https://github.com/Zven94/OOP-school-library/assets/78251927/35ce4800-9720-4f84-9d59-d8f699c1fd61)



Your App (Please remove the option `Classroom`)
![image](https://github.com/Zven94/OOP-school-library/assets/78251927/bec9ff81-aba3-4d37-8864-77bc15316ae5)

**List all people**
![image](https://github.com/Zven94/OOP-school-library/assets/78251927/5ab1446c-2441-48d2-96f4-f5bd1175784f)

Kindly ensure that the printed result corresponds with the data input. i.e. `name: Dafe` and `age: 6` 🤓

Reply:
Hi @abdulhamiid  but the classroom was a requeriment from the previus project. Do i have to delete it?

Reply:
Hi, I see you removed the classroom option, yes, this was not required in this milestone 👍🏿